aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -118,6 +118,51 @@ SparseTensor& log1p_sparse_(SparseTensor& t) {
   return log1p_out_sparse(t, t);
 }
 
+// --------------------------------------------------------------------
+// neg(SparseTensor)
+// --------------------------------------------------------------------
+
+SparseTensor& neg_out_sparse(SparseTensor& r, const SparseTensor& t) {
+  AT_ASSERT(r.is_sparse());
+  AT_ASSERT(t.is_sparse());
+
+  // copy_sparse_ does not perform the copy if it is the same tensor
+  copy_sparse_to_sparse_(r, t);
+  r._values().neg_();
+  return r;
+}
+
+SparseTensor& neg_sparse_(SparseTensor& t) {
+  return neg_out_sparse(t, t);
+}
+
+// --------------------------------------------------------------------
+// asin(SparseTensor)
+// --------------------------------------------------------------------
+
+// In-place asin on uncoalesced tensors is not supported due non-linear type of operation.
```

Comment:
"is not supported due non-linear type of operation." -> "is not supported since the operation is not a linear map." 

Technically additivity would be a sufficient property for operations on uncoalesced tensors, but I would be very impressed if PyTorch ever implemented an additive function that was not also a linear map.